.github/workflows/go.yml

```diff
@@ -23,7 +26,7 @@ jobs:
     - name: Set up Go 1.x
       uses: actions/setup-go@v2
       with:
-        go-version: 1.17
+        go-version: 1.18
```

Comment:
Is this related to the ruby problem? I think there's a bigger gap now between the go version in our dockerfiles and in the ci (v1.16)

Reply:
I'm not sure what you mean by ci, isn't this file the one that dictates what used by ci? I upgraded to this version to match the one found in latest Fedora 'golang-bin' rpm.

Reply:
I mean that this file, for our CI, now specifies go 1.18, whereas the Dockerfile still specifies go 1.16.

Reply:
I also updated the Dockerfile to use the default golang-bin package from Fedora (currently 1.18).

Reply:
@jasinner sorry, I'm not try to hold this PR up, overall this is seems to be a helpful change. I'll try to clarify my intention in the original comment:

a) is updating the Go version a required part of this change, to fix the ruby issue?

b) If it is, can we try to make the go version consistent in all places referenced in this repo?

c) If it is not, would it be easier to revert to Go version changes, and just fix the ruby problem?

Reply:
I can take out the golang related changes if you prefer, they are not required to fix the issues described in issue/19. The reason I added it was because the README says to install golang-bin, which if you do that with the latest Fedora image you got 1.18. Therefore went around updating the go version in other places to make that consistent.

Reply:
Ok thanks for the explanation!

It seems deplist is not very consistent in it's usage of Go versions at present. I see four different places:
* ci: 1.17
* go.mod: 1.17
* dockerfile: 1.16
* readme: undefined (though perhaps implicitly following go.mod?)

I think we should try to bring these all up to the same version at once, but should probably be done separately to this PR, if it's not needed. If you can undo the readme change, I'm happy to merge.